frontend/src/components/admin/AssignedStoryTranslationsTable.tsx

```diff
@@ -0,0 +1,176 @@
+import { getLevelVariant } from "../../utils/StatusUtils";
+import convertStageTitleCase from "../../utils/StageUtils";
+import { convertLanguageTitleCase } from "../../utils/LanguageUtils";
+
+export type AssignedStoryTranslationsTableProps = {
+  storyTranslations: StoryTranslation[];
+  setStoryTranslations: (newStoryTranslations: StoryTranslation[]) => void;
+  userId: number;
+};
+
+const AssignedStoryTranslationsTable = ({
+  storyTranslations,
+  setStoryTranslations,
+  userId,
+}: AssignedStoryTranslationsTableProps) => {
+  const [isAscendingTitle, setIsAscendingTitle] = useState(true);
+  const [isAscendingRole, setIsAscendingRole] = useState(true);
+  const [isAscendingLanguage, setIsAscendingLanguage] = useState(true);
+  const [isAscendingStage, setIsAscendingStage] = useState(true);
+
+  const getRole = (translation: StoryTranslation) => {
+    if (userId === translation.translatorId) {
+      return "Translator";
+    }
+    if (userId === translation.reviewerId) {
+      return "Reviewer";
+    }
+    return "Error";
+  };
+
+  const titleSort = (t1: StoryTranslation, t2: StoryTranslation) =>
+    isAscendingTitle
+      ? t1.title.localeCompare(t2.title)
+      : t2.title.localeCompare(t1.title);
+
+  const roleSort = (t1: StoryTranslation, t2: StoryTranslation) =>
+    isAscendingRole
+      ? getRole(t1).localeCompare(getRole(t2))
+      : getRole(t2).localeCompare(getRole(t1));
+
+  const languageSort = (t1: StoryTranslation, t2: StoryTranslation) =>
+    isAscendingLanguage
+      ? t1.language.localeCompare(t2.language)
+      : t2.language.localeCompare(t1.language);
+
+  const stageSort = (t1: StoryTranslation, t2: StoryTranslation) =>
+    isAscendingStage
+      ? t1.stage.localeCompare(t2.stage)
+      : t2.stage.localeCompare(t1.stage);
+
+  const sort = (field: String) => {
+    const newStoryTranslations = [...storyTranslations];
+    switch (field) {
+      case "title": {
+        setIsAscendingTitle(!isAscendingTitle);
+
+        newStoryTranslations.sort(titleSort);
+        break;
+      }
+      case "role": {
+        setIsAscendingRole(!isAscendingRole);
+
+        newStoryTranslations.sort(roleSort);
+        break;
+      }
+      case "language": {
+        setIsAscendingLanguage(!isAscendingLanguage);
+
+        newStoryTranslations.sort(languageSort);
+        break;
+      }
+      case "stage": {
+        setIsAscendingStage(!isAscendingStage);
+
+        newStoryTranslations.sort(stageSort);
+        break;
+      }
+      default: {
+        break;
+      }
+    }
+    setStoryTranslations(newStoryTranslations);
+  };
+
+  const tableBody = storyTranslations.map(
+    (translation: StoryTranslation, index: number) => (
+      <Tr
+        borderBottom={
+          index === storyTranslations.length - 1 ? "1em solid transparent" : ""
+        }
+        key={`${translation?.storyTranslationId}`}
+      >
+        <Td>{translation.title}</Td>
```

Comment:
nit can you link the title like so https://github.com/uwblueprint/planet-read/blob/main/frontend/src/components/admin/StoryTranslationsTable.tsx#L79-L84